pkg/hubble/cell/hubble.go

```diff
@@ -0,0 +1,619 @@
+			WithField("reasons", h.config.K8sDropEventsReasons).
+			Info("Starting packet drop events emitter")
+
+		dropEventEmitter := dropeventemitter.NewDropEventEmitter(
+			h.config.K8sDropEventsInterval,
+			h.config.K8sDropEventsReasons,
+			h.clientset,
+			h.k8sWatcher,
+		)
+
+		observerOpts = append(observerOpts,
+			observeroption.WithOnDecodedFlowFunc(func(ctx context.Context, flow *flowpb.Flow) (bool, error) {
+				err := dropEventEmitter.ProcessFlow(ctx, flow)
+				if err != nil {
+					h.log.WithError(err).Error("Failed to ProcessFlow in drop events handler")
+				}
+				return false, nil
+			}),
+		)
+	}
+
+	// fill in the local node information after the dropEventEmitter logique,
+	// but before anything else (e.g. metrics).
+	localNodeWatcher, err := observer.NewLocalNodeWatcher(ctx, h.nodeLocalStore)
+	if err != nil {
+		h.log.WithError(err).Error("Failed to retrieve local node information")
+		return
+	}
+	observerOpts = append(observerOpts, observeroption.WithOnDecodedFlow(localNodeWatcher))
+
+	grpcMetrics := grpc_prometheus.NewServerMetrics()
+	var metricsTLSConfig *certloader.WatchedServerConfig
+	if h.config.EnableMetricsServerTLS {
+		metricsTLSConfigChan, err := certloader.FutureWatchedServerConfig(
+			h.log.WithField("config", "hubble-metrics-server-tls"),
+			h.config.MetricsServerTLSClientCAFiles,
+			h.config.MetricsServerTLSCertFile,
+			h.config.MetricsServerTLSKeyFile,
+		)
+		if err != nil {
+			h.log.WithError(err).Error("Failed to initialize Hubble metrics server TLS configuration")
+			return
+		}
+		waitingMsgTimeout := time.After(30 * time.Second)
+		for metricsTLSConfig == nil {
+			select {
+			case metricsTLSConfig = <-metricsTLSConfigChan:
+			case <-waitingMsgTimeout:
+				h.log.Info("Waiting for Hubble metrics server TLS certificate and key files to be created")
+			case <-ctx.Done():
+				h.log.WithError(ctx.Err()).Error("Timeout while waiting for Hubble metrics server TLS certificate and key files to be created")
+				return
+			}
+		}
+		go func() {
+			<-ctx.Done()
+			metricsTLSConfig.Stop()
+		}()
+	}
+
+	var srv *http.Server
+	if h.config.MetricsServer != "" {
+		h.log.WithFields(logrus.Fields{
+			"address": h.config.MetricsServer,
+			"metrics": h.config.Metrics,
+			"tls":     h.config.EnableMetricsServerTLS,
+		}).Info("Starting Hubble Metrics server")
+
+		err := metrics.InitMetrics(metrics.Registry, api.ParseStaticMetricsConfig(h.config.Metrics), grpcMetrics)
+		if err != nil {
+			h.log.WithError(err).Error("Unable to setup metrics: %w", err)
+			return
+		}
+
+		srv = &http.Server{
+			Addr:    h.config.MetricsServer,
+			Handler: nil,
+		}
+		metrics.InitMetricsServerHandler(srv, metrics.Registry, h.config.EnableOpenMetrics)
+
+		go func() {
+			if err := metrics.StartMetricsServer(srv, h.log, metricsTLSConfig, grpcMetrics); err != nil && !errors.Is(err, http.ErrServerClosed) {
+				h.log.WithError(err).Error("Hubble metrics server encountered an error")
+				return
+			}
+		}()
```

Comment:
We'll likely want to implement a metrics cell in a follow-up PR. And probably others as the launch method is still massive and starts various subsystems.

Reply:
Wholeheartly agree, I mentioned in the introduction commit that splitting the Hubble components into smaller cells is left for a later time as this PR goal is to introduce a "top level Hubble cell" to bootstrap the process.